darwin/main.m

```diff
@@ -118,20 +119,26 @@ - (BOOL)applicationShouldTerminateAfterLastWindowClosed:(NSApplication *)app
 		[realNSApp() setMainMenu:[appDelegate().menuManager makeMenubar]];
 
 		setupFontPanel();
-
-		return NULL;
 	}
+
+	globalPool = [[NSAutoreleasePool alloc] init];
```

Comment:
It was unnecessary to have the global pool created before the local pool, since the local pool covers all autoreleased objects in that function. The global pool should then be created right before exit of uiInit, and released right at the beginning of the uiUnint to catch all autoreleased objects created after uiInit. NSApplication run will create its own pools for each run loop, so we don't need to worry about pools for functions called after uiMain.

Moving this actually uncovered the over-release I found with pageID and view in tabPage.

You could also get rid of NSAutoreleasePool and enclose all uiXxx functions with @autoreleasepool. Could be doable with some type of macro possibly. NSAutoreleasePool is a no-go for ARC, but @autoreleasepools are fine.